### PR TITLE
Adds the new replace flow to the audio block

### PR DESCRIPTION
**Pull request by draganescu**
## Description
There is a new replace flow offered by a component which is currently implemented in the Image Block. This PR implements the new flow in the Audio Block.

## How has this been tested?
Tested locally.

## Screenshots

Screenshot 2019-12-16 at 10 41 36

### CODE REVIEW

File: packages/block-library/src/audio/edit.js [hunk 1]
```diff
@@ -118,24 +108,19 @@ class AudioEdit extends Component {
...
 				// in this case there was an error and we should continue in the editing state
 				// previous attributes should be removed because they may be temporary blob urls
 				setAttributes( { src: undefined, id: undefined } );
-				switchToEditing();
 				return;
 			}
 			// sets the block's attribute and updates the edit component from the
 			// selected media, then switches off the editing UI
 			setAttributes( { src: media.url, id: media.id } );
 			this.setState( { src: media.url, editing: false } );
```
Comment (Soean) [inline packages/block-library/src/audio/edit.js]: We removed the `editing` state in this PR, the state `src` was removed in #6820, so we could delete this line. 
```suggestion
```

**Review by Soean — CHANGES_REQUESTED** [review]
The replacement flow works great. I added just a little suggestion.

**Comment by Soean** [comment]
This PR fixes #16048

File: packages/block-library/src/audio/edit.js (hunk 1 above)
Comment (draganescu) [reply packages/block-library/src/audio/edit.js]: Good catch, I removed the line in [d7681fa](https://github.com/WordPress/gutenberg/pull/19158/commits/d7681fa9768deef4eabc3a3dbbd9cbe29b482aad).

**Review by Soean — APPROVED** [review]
Thanks, code looks good and replacement works.

**Pull request merged**